geemap/common.py

```diff
@@ -16167,3 +16167,160 @@ def is_on_aws():
         if item.endswith(".aws") or "ec2-user" in item:
             on_aws = True
     return on_aws
+
+
+def google_map_tiles(
```

Comment:
Having one function that can return either a single `TileProvider` or a dictionary of `TileProvider` objects ends up overcomplicating the code. Consider breaking this up into:

1) a function to create the dictionary of `TileProvider` objects, and
2) a `TileProvider` subclass, `GoogleMapsTileLayer`.

Reply:
I have added a `GoogleMapsTileProvider` class and a `google_map_tiles` function, both can be found in the `basemaps.py`